src/OSPSuite.Infrastructure.Import/Core/DataFormat/AbstractColumnsDataFormat.cs

```diff
@@ -218,11 +218,18 @@ public IEnumerable<ParsedDataSet> Parse(DataSheet dataSheet, ColumnInfoCache col
          if (missingColumns.Any())
             throw new MissingColumnException(dataSheet.SheetName, missingColumns);
 
-         var groupingCriteria =
+         //we first isolate the mapping parameters
+         var mappingCriteria =
             Parameters
-               .Where(p => p.IsGroupingCriterion())
+               .Where(p => p.IsGroupingCriterion() && !p.IsAnImplementationOf<GroupByDataFormatParameter>())
```

Comment:
Can you use the OrderBy to preserve this as a single step (or possibly OrderByDescending).
```
         var mappingCriteria =
            Parameters
               .Where(p => p.IsGroupingCriterion()).OrderBy(p => p.IsAnImplementationOf<GroupByDataFormatParameter>())
               .Select(p => p.ColumnName);

```

Then you wouldn't need to perform the union as step 2. Although, you still need the comment to describe why you are ordering at all.

Reply:
good suggestion @rwmcintosh. I have adjusted the code.